src/main/java/com/influxdb/v3/client/internal/NanosecondConverter.java

```diff
@@ -111,4 +118,53 @@ public static BigInteger convert(final Instant instant, final WritePrecision pre
 
         return FROM_NANOS.get(precision).apply(nanos);
     }
+
+    /**
+     * Convert Long or LocalDateTime to timestamp nanosecond.
+     *
+     * @param value  the time in Long or LocalDateTime
+     * @param field the arrow field metadata
+     * @return the time in nanosecond
+     */
+    public static BigInteger getTimestampNano(@Nonnull final Object value, @Nonnull final Field field) {
```

Comment:
The result can be nullable:

```suggestion
    @Nullable
    public static BigInteger getTimestampNano(@Nonnull final Object value, @Nonnull final Field field) {
```